packages/idx/src/idx.d.ts

```diff
@@ -33,9 +33,15 @@ type DeepRequired<T> = T extends any[]
 
 /**
  * UnboxDeepRequired
- * Unbox type wraped with DeepRequired
+ * Unbox type warped with DeepRequired
  */
-type UnboxDeepRequired<T> = T extends DeepRequired<infer R> ? R : T;
+type UnboxDeepRequired<T> = T extends DeepRequired<any>
+  ? T extends DeepRequiredArray<infer R>
+    ? Array<R>
+    : T extends (...args: infer A) => infer R
+      ? (...args: A) => UnboxDeepRequired<R>
+      : T extends DeepRequiredObject<infer R> ? R : T
+  : T;
```

Comment:
on second thought, we can remove the `extends DeepRequired<any>` part.
here's the version i have working well locally.
```
type UnboxDeepRequired<T> = T extends DeepRequiredArray<infer R>
   ? Array<R>
   : T extends (...args: infer A) => infer R
     ? (...args: A) => UnboxDeepRequired<R>
     : T extends DeepRequiredObject<infer R> ? R : T;
```

Reply:
We can move it to type argument qualifier 

```ts
type UnboxDeepRequired<
  T extends DeepRequired<any>
> = T extends DeepRequiredArray<infer R>
  ? Array<R>
  : T extends (...args: infer A) => infer R
    ? (...args: A) => UnboxDeepRequired<R>
    : T extends DeepRequiredObject<infer R> ? R : T;
```

We want to make sure `UnboxDeepRequired` only accepts `DeepsRequired` types as type argument